### PR TITLE
gen/namespace: don't allow reserved names

### PR DESCRIPTION
**Pull request by abhinav**

### CODE REVIEW

File: gen/reserved.go [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package gen
+
+var _reservedNames = make(map[string]struct{})
+
+func init() {
+	reservedNames := []string{
```
Comment (prashantv) [inline gen/reserved.go]: put a comment about where you got the list.

Comment (prashantv) [inline gen/reserved.go]: also, their list skips `error` but that should be reserved too, see
https://github.com/uber/tchannel-go/blob/dev/thrift/thrift-gen/names.go#L28

**Comment by prashantv** [comment]
lgtm, make sure to add `error`

File: gen/reserved.go (hunk 1 above)
Comment (abhinav) [inline gen/reserved.go]: :+1:

**Pull request merged**